### ISSUE TITLE
SqlCeScripting40

### ISSUE THREAD

**Issue by elenaaralla** [ErikEJ/SqlCeToolbox#886]
Hi,

hi tried to export a sqlce database to sqlite using SqlCeScripting40.dll and the example code in this article:

http://erikej.blogspot.com/2013/03/sql-server-compact-code-snippet-of-week.html

but I had some problem with PK and FK conversion, i.e. 

primary key that on CE are of type **int** become INTEGER in script
primary key that on CE are of type **bigint** become INTEGER in script too

FK to this two primary key, in script will be **int** the first one and **bigint** the second one.

When import script on sqlite (with sqlite3.exe) the INTEGER type become a **bigint** , so the first Foreign Key is between int and bigint and Entity Framework, when i try to generate an edmx file,  reports me an error due to the fact that the foreign key is between two different types.

I manually correct the type of fields involved in the wrong FK, but i need to automate this process.

I'm using Visual Studio 2017, Entity Framework 6, SqlCeScripting version 3.5.2.79

this is my code:

```
            using (IRepository repository = new DB4Repository(@"Data Source=D:\WORKING\GPEC_SQLite\database\test.sdf"))
            {
                IGenerator generator = new Generator4(repository, @"D:\WORKING\GPEC_SQLite\database\test.sqlite");
                generator.ScriptDatabaseToFile(Scope.SchemaDataSQLite);
            }

            using (IRepository repository = new DB4Repository(@"Data Source=D:\WORKING\GPEC_SQLite\database\test.sdf"))
            {
                IGenerator generator = new Generator4(repository, @"D:\WORKING\GPEC_SQLite\database\test.sqlce");
                generator.ScriptDatabaseToFile(Scope.SchemaData);
            }
```

in the attached zip file there are:
test.sdf database
generated script for SqlCE
generated script for SQLite

[script_and_db.zip](https://github.com/ErikEJ/SqlCeToolbox/files/6158379/script_and_db.zip)

**Comment by ErikEJ** [comment]
I am not sure that this scenario is supported, but agree that both FK properties should also be INTEGER.

**Comment by ErikEJ** [comment]
How are you generating an EDMX file?

**Comment by elenaaralla** [comment]
> How are you generating an EDMX file?

I'm not sure I understand the question, anyway, this is what I did: I generated the edmx by adding a new "AdoNet Entity Data Model"; in the attached zip, the images of the steps I made.
The result is that the tables are not mapped due to this error:

Error 13101: The types of all properties in the Dependent Role of a referential constraint must be the same as the types of the corresponding properties in the Principal Role. Property type 'idA' in entity 'table_c' does not match property type 'Id' in entity 'table_a' in referential constraint 'FK_table_c_1_0'.

Could the sample project be useful to you?

[Archivio.zip](https://github.com/ErikEJ/SqlCeToolbox/files/6290234/Archivio.zip)
